File: 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;
    int length = strlen(text);

    for( int i = 0; i < length; i++){
        if(text[i] == ' '){
            words++;
        }
        else if(text[i] == '.' || text[i] == '!' || text[i] == '?'){
            sentences++;
        }
        else if(isupper(text[i]) || islower(text[i])){
            letters++;
        }
    }

    double average_n = ((double)letters/(double)words)*100;
    double average_s = ((double)sentences/(double)words)*100;

    int index = round(0.0588 * average_n - 0.296 * average_s - 15.8);

    if( index >= 16){
        printf("Grade 16+\n");
    }
    else if( index < 1){
        printf("Before Grade 1\n");

    }
    else{
        printf("Grade %d\n",index);
    }
}

File: 6/sentimental-readability/readability.py
from cs50 import get_string
import string

sentence = get_string("Text: ")

# Declare variables
letters = 0
words = 1
sentences = 0

for letter in sentence:
    if letter == "!" or letter == "?" or letter == ".":
        sentences += 1
    elif letter in string.punctuation:
        continue
    elif letter in string.whitespace:
        words += 1
    else:
        letters += 1

wordFactor = words / 100
lettersPer100 = letters / wordFactor
sentencesPer100 = sentences / wordFactor
liauIndex = round((0.0588 * lettersPer100) - (0.296 * sentencesPer100) - 15.8)

if liauIndex > 16:
    print("Grade 16+")
elif liauIndex < 1:
    print("Before Grade 1")
else:
    print(f"Grade {liauIndex}")